src/jld_types.jl

```diff
@@ -523,11 +560,12 @@ end
 ## Common functions for all non-special types (including gen_h5convert)
 
 # Whether this datatype should be stored as opaque
-isopaque(t::(Type...)) = isa(t, ())
-isopaque(t::DataType) = isempty(t.names)
+isopaque(t::TupleType) = t == EMPTY_TUPLE_TYPE
+# isopaque(t::DataType) = isempty(fieldnames(t))
+isopaque(t::DataType) = isa(t, TupleType) ? t == EMPTY_TUPLE_TYPE : isempty(fieldnames(t))
```

Comment:
This is how I noticed JuliaLang/julia#10978, but I think `isopaque(t::TupleType)` should be called if `isa(t, TupleType)`, and not this method.